instance/management/commands/instance_statistics_csv.py

```diff
@@ -0,0 +1,294 @@
+            """Helper to pass to capture_playbook_output()."""
+            self.stderr.write(self.style.SUCCESS(line))
+        log_line.info = log_line
+        log_line.error = log_line
+
+        # Launch the collect_elasticsearch_data playbook, which places a file into the `playbook_output_dir`
+        # on this host.
+        ansible.capture_playbook_output(
+            requirements_path=os.path.join(
+                os.path.dirname(playbook_path),
+                'requirements.txt'
+            ),
+            inventory_str=inventory,
+            vars_str=(
+                'local_output_dir: {output_dir}\n'
+                'remote_output_filename: /tmp/activity_report\n'
+                'server_name_prefix: {server_name_prefix}\n'
+                'start_date: {start_date}\n'
+                'end_date: {end_date}'
+            ).format(
+                output_dir=playbook_output_dir,
+                server_name_prefix=name_prefix,
+                start_date=start_date,
+                end_date=end_date
+            ),
+            playbook_path=playbook_path,
+            username=settings.OPENSTACK_LOGS_SERVER_SSH_USERNAME,
+            logger_=log_line,
+        )
+
+    def get_appserver_statistics(self, playbook_output_dir, name_prefix, public_ip):
+        """ Execute the collect_appserver_data playbook to gather statistics """
+        inventory = '[apps]\n{server}'.format(server=public_ip)
+        playbook_path = os.path.join(
+            settings.SITE_ROOT,
+            'playbooks/collect_activity/collect_activity.yml'
+        )
+
+        def log_line(line):
+            """Helper to pass to capture_playbook_output()."""
+            self.stderr.write(self.style.SUCCESS(line))
+        log_line.info = log_line
+        log_line.error = log_line
+
+        # Launch the collect_appserver_data playbook, which places a file into the `playbook_output_dir`
+        # on this host.
+        ansible.capture_playbook_output(
+            requirements_path=os.path.join(os.path.dirname(playbook_path), 'requirements.txt'),
+            inventory_str=inventory,
+            vars_str=(
+                'local_output_dir: {output_dir}\n'
+                'local_output_filename: user_statistics\n'
+                'remote_output_filename: /tmp/activity_report\n'
+                'server_name_prefix: {server_name_prefix}\n'
+                'extra_script_arguments: --skip-hit-statistics'
+            ).format(
+                output_dir=playbook_output_dir,
+                server_name_prefix=name_prefix
+            ),
+            playbook_path=playbook_path,
+            username=settings.OPENSTACK_SANDBOX_SSH_USERNAME,
+            logger_=log_line,
+        )
+
+    def collect_instance_statistics(
+            self,
+            out,
+            domain_name,
+            start_date,
+            end_date
+    ):  # pylint: disable=too-many-locals
+        """Generate the instance statistics CSV."""
```

Comment:
@pcockwell, I'm just wondering if it will be cleaner to add a flag to the existing management command to restrict it to a single instance, run it and use its output and just add the elasticsearch data?

Reply:
I don't think it's quite that simple because the current command compiles data across all instances/appservers. I think if we want to enable the ability to get data from elasticsearch for all instances, we should extend the current command, and eventually deprecate the existing one. I don't think it really makes sense to try and mangle the two together.

Reply:
@pcockwell, I am not sure if you understood the point that I was making - one appserver or a hundred appservers, the code that is used to collect the usage data (counts of users, courses etc.) is exactly the same. And that is why I was asking if we can leverage the existing playbook.

I checked the code of the `activity_csv` management command it looks fairly straightforward to add the functionality to restrict the playbook run to a smaller subset or to an instance by adding a flag/option.

The `get_active_servers()` function [here](https://github.com/open-craft/opencraft/blob/master/instance/management/commands/activity_csv.py#L74) is used to get the list of active appservers which is then used to generate the inventory for running the playbook on.

So can't we invoke the existing command, restricting its scope to the instance/appserver(s) related to the current management command, get the CSV, parse it and then add the log data summary from elasticsearch?